pvdeg/utilities.py

```diff
@@ -1575,3 +1534,170 @@ def read_material_property(
             for k, v in material_dict.items()
         }
     return material_dict
+
+
+def gids_dataset_to_coords_dataset(ds_gids: xr.Dataset, meta_df: pd.DataFrame):
+    """
+    Convert dataset gids to gridded latitude, longitude dataset.
+    Maintains all other coordiantes.
+
+    Aims to support advanced workflows where pvdeg.geospatial.analysis is applied twice.
+
+    Parameters
+    ----------
+    ds_gids : xr.Dataset
+        dataset with "gid" dimension/coords
+    meta_df : pd.DataFrame
+        metadata pandas dataframe containing gid index, and latitude
+        and longitude columns
+
+    Returns
+    -------
+    coords_ds: xr.Dataset
+        dataset with "latitude", "longitude" dimensions/coords in
+        addition to original coords not including "gids"
+    """
+
+    meta_df = meta_df.loc[ds_gids.gid]
+
+    stacked = ds_gids.drop(["gid"])
+
+    mindex_obj = pd.MultiIndex.from_arrays(
+        [meta_df["latitude"], meta_df["longitude"]], names=["latitude", "longitude"]
+    )
+    mindex_coords = xr.Coordinates.from_pandas_multiindex(mindex_obj, "gid")
+    stacked = stacked.assign_coords(mindex_coords)
+
+    stacked = stacked.drop_duplicates("gid")
+    res = stacked.unstack("gid")
+    return res
+
+
+def _load_gcr_from_config(config_files: dict):
+    """
+    dictionary containg 'pv' key
+    """
+
+    import json
+
+    with open(config_files["pv"], "r") as fp:
+        data = json.load(fp)
+
+    return data["subarray1_gcr"]
+
+
+def optimal_gcr_pitch_bifacial_fixed_tilt(
+    latitude: float, cw: float = 2
+) -> tuple[float, float]:
+    """
+    Determine optimal gcr and pitch for fixed tilt systems according
+    to latitude and optimal GCR parameters for fixed tilt bifacial systems.
+
+    .. math::
+
+        GCR = \frac{P}{1 + e^{-k(\alpha - \alpha_0)}} + GCR_0
+
+    Inter-row energy yield loss 5% Bifacial Parameters:
+
+    +-----------+--------+-----------+
+    | Parameter | Value  | Units     |
+    +===========+========+===========+
+    | P         | 0.560  | unitless  |
```

Comment:
This value should be negative (see the paper: https://doi.org/10.1016/j.solener.2023.04.038)

```suggestion
    | P         | -0.560  | unitless  |
```